gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/transactions/TransactionDUnit.java

```diff
@@ -145,6 +146,140 @@ public static void resetObservers() {
+                  otherSt.execute("insert into trade.t1 values (" + i + "," + i+")");
+                }
+                while (!executed[0]) {
+                  synchronized (waitLock) {
+                    executed[1] = true;
+                    waitLock.wait();
+                  }
+                }
+                Misc.getI18NLogWriter().info(LocalizedStrings.DEBUG, "Going to inesrt for one bucket");
+
+                PartitionedRegion r = (PartitionedRegion)Misc.getRegionForTable("TRADE.T1", true);
+
+                for(int i:r.getLocalBucketsListTestOnly()) {
+                  Misc.getI18NLogWriter().info(LocalizedStrings.DEBUG, "Going to inesrt for bucket id " + i);
+                  otherSt.execute("insert into trade.t1 values (" + (i + 113 * 4) + ", 20)");
+                }
+
+                otherConn.commit();
+              } catch (Exception e) {
+                otherConn.rollback();
+                e.printStackTrace();
+              }
+            } catch (Exception e) {
+              e.printStackTrace();
+            } finally {
+            }
+          }
+        });
+
+        otherTx.start();
+        while (!executed[1]) {
+          try {
+            Thread.sleep(100);
+          } catch (InterruptedException e) {
+            e.printStackTrace();
+          }
+        }
+        final TransactionObserver observer = new TransactionObserverAdapter() {
+
+          @Override
+          public void beforePerformOp(TXStateProxy tx) {
+            Misc.getI18NLogWriter().info(LocalizedStrings.DEBUG, "before Op : Waiting for 10 sec");
+            try {
+              Thread.sleep(60000);
+              Thread.sleep(60000);
+            } catch (InterruptedException e) {
+              e.printStackTrace();
+            }
+            Misc.getI18NLogWriter().info(LocalizedStrings.DEBUG, "before Op : Waiting for 10 sec over.");
+          }
+        };
+        final GemFireCacheImpl cache = GemFireCacheImpl.getInstance();
+        if (cache != null) {
+          for (TXStateProxy tx : cache.getTxManager().getHostedTransactionsInProgress()) {
+            Misc.getI18NLogWriter().info(LocalizedStrings.DEBUG,"attaching observer to " + tx);
+
+            tx.setObserver(observer);
+          }
+        }
+
+        /*invokeInEveryVM(new SerializableRunnable() {
+          @Override
+          public void run() {
+            failed = false;
+
+          }
+        });*/
+
+
+        synchronized (waitLock) {
+          executed[0] = true;
+          waitLock.notify();
+        }
+       /* try {
+          otherTx.join();
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }*/
+      }
+    });
+    Thread.sleep(10000);
+    stopVMNum(-2);
+    stopVMNum(-3);
+
+    for(int i=700;i<1000;i++) {
+      try {
+        st.execute("insert into trade.t1 values (639, 20)");
+      }
+      catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+
+    Thread.sleep(300000);
```

Comment:
@suranjan Why are there such large sleeps here and above (> 7 mins)? Apart from what Neeraj mentioned, precheckin time has increased substantially.

Reply:
Have reduced the sleeps for now. If required we can discuss what can be done for testing this cleanly.